diff -uprN ./arch/x86/ctx.h /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.h
--- ./arch/x86/ctx.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.h	2016-11-30 18:28:48.000000000 +0900
@@ -35,10 +35,6 @@
 #ifndef CTX_H
 #define CTX_H

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 /*
  * CPU context registers
  */
@@ -57,8 +53,5 @@ struct ctx {
 void
 ctx_switch(struct ctx *new_ctx, struct ctx *curr_ctx);

-#ifdef __cplusplus
-} /* extern C */
-#endif

 #endif /* RTE_CTX_H_ */
Binary files ./arch/x86/ctx.o and /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.o differ
diff -uprN ./lthread_api.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_api.h
--- ./lthread_api.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_api.h	2016-11-30 18:28:48.000000000 +0900
@@ -124,10 +124,6 @@
 #ifndef LTHREAD_H
 #define LTHREAD_H

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <sys/socket.h>
 #include <fcntl.h>
@@ -833,8 +829,4 @@ int lthread_cond_signal(struct lthread_c
   */
 int lthread_cond_broadcast(struct lthread_cond *c);

-#ifdef __cplusplus
-} /* extern "C" */
-#endif
-
 #endif				/* LTHREAD_H */
diff -uprN ./lthread_cond.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.h
--- ./lthread_cond.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.h	2016-11-30 18:28:48.000000000 +0900
@@ -62,10 +62,6 @@
 #ifndef LTHREAD_COND_H_
 #define LTHREAD_COND_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include "lthread_queue.h"

 #define MAX_COND_NAME_SIZE 64
@@ -78,8 +74,4 @@ struct lthread_cond {
 	uint64_t diag_ref;	/* optional ref to user diag data */
 } __rte_cache_aligned;

-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_COND_H_ */
Binary files ./lthread_cond.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.o differ
diff -uprN ./lthread-cpp.patch /home/slank/dpdk-16.07/examples/performance-thread/common/lthread-cpp.patch
--- ./lthread-cpp.patch	2017-03-14 01:47:40.126985537 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread-cpp.patch	1970-01-01 09:00:00.000000000 +0900
@@ -1,71 +0,0 @@
-diff -uprN ./arch/x86/ctx.h /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.h
---- ./arch/x86/ctx.h	2017-03-14 01:43:01.073260218 +0900
-+++ /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.h	2016-11-30 18:28:48.000000000 +0900
-@@ -35,10 +35,6 @@
- #ifndef CTX_H
- #define CTX_H
-
--#ifdef __cplusplus
--extern "C" {
--#endif
--
- /*
-  * CPU context registers
-  */
-@@ -57,8 +53,5 @@ struct ctx {
- void
- ctx_switch(struct ctx *new_ctx, struct ctx *curr_ctx);
-
--#ifdef __cplusplus
--} /* extern C */
--#endif
-
- #endif /* RTE_CTX_H_ */
-Binary files ./arch/x86/ctx.o and /home/slank/dpdk-16.07/examples/performance-thread/common/arch/x86/ctx.o differ
-diff -uprN ./lthread_api.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_api.h
---- ./lthread_api.h	2017-03-14 01:43:01.073260218 +0900
-+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_api.h	2016-11-30 18:28:48.000000000 +0900
-@@ -124,10 +124,6 @@
- #ifndef LTHREAD_H
- #define LTHREAD_H
-
--#ifdef __cplusplus
--extern "C" {
--#endif
--
- #include <stdint.h>
- #include <sys/socket.h>
- #include <fcntl.h>
-@@ -833,8 +829,4 @@ int lthread_cond_signal(struct lthread_c
-   */
- int lthread_cond_broadcast(struct lthread_cond *c);
-
--#ifdef __cplusplus
--} /* extern "C" */
--#endif
--
- #endif				/* LTHREAD_H */
-diff -uprN ./lthread_cond.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.h
---- ./lthread_cond.h	2017-03-14 01:43:01.073260218 +0900
-+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.h	2016-11-30 18:28:48.000000000 +0900
-@@ -62,10 +62,6 @@
- #ifndef LTHREAD_COND_H_
- #define LTHREAD_COND_H_
-
--#ifdef __cplusplus
--extern "C" {
--#endif
--
- #include "lthread_queue.h"
-
- #define MAX_COND_NAME_SIZE 64
-@@ -78,8 +74,4 @@ struct lthread_cond {
- 	uint64_t diag_ref;	/* optional ref to user diag data */
- } __rte_cache_aligned;
-
--#ifdef __cplusplus
--} /* extern C */
--#endif
--
- #endif				/* LTHREAD_COND_H_ */
-Binary files ./lthread_cond.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_cond.o differ
diff -uprN ./lthread_diag_api.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_diag_api.h
--- ./lthread_diag_api.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_diag_api.h	2016-11-30 18:28:48.000000000 +0900
@@ -33,10 +33,6 @@
 #ifndef LTHREAD_DIAG_API_H_
 #define LTHREAD_DIAG_API_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <inttypes.h>

@@ -326,9 +322,4 @@ lthread_cond_diag_ref(struct lthread_con
 uint64_t
 lthread_mutex_diag_ref(struct lthread_mutex *m);

-
-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_DIAG_API_H_ */
diff -uprN ./lthread_diag.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_diag.h
--- ./lthread_diag.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_diag.h	2016-11-30 18:28:48.000000000 +0900
@@ -34,10 +34,6 @@
 #ifndef LTHREAD_DIAG_H_
 #define LTHREAD_DIAG_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <inttypes.h>

@@ -132,9 +128,5 @@ extern uint64_t diag_mask;

 #define DIAG_USED __rte_unused

-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_DIAG */
 #endif				/* LTHREAD_DIAG_H_ */
Binary files ./lthread_diag.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_diag.o differ
diff -uprN ./lthread.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread.h
--- ./lthread.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread.h	2016-11-30 18:28:48.000000000 +0900
@@ -62,10 +62,6 @@
 #ifndef LTHREAD_H_
 #define LTHREAD_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <rte_per_lcore.h>

 #include "lthread_api.h"
@@ -100,8 +96,4 @@ _lthread_init(struct lthread *lt,

 void _lthread_set_stack(struct lthread *lt, void *stack, size_t stack_size);

-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_H_ */
diff -uprN ./lthread_int.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_int.h
--- ./lthread_int.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_int.h	2016-11-30 18:28:48.000000000 +0900
@@ -62,10 +62,6 @@
 #include <lthread_api.h>
 #define LTHREAD_INT_H

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <stdint.h>
 #include <sys/time.h>
 #include <sys/types.h>
@@ -201,8 +197,4 @@ struct lthread {
 	uint64_t diag_ref;			/* ref to user diag data */
 } __rte_cache_aligned;

-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_INT_H */
diff -uprN ./lthread_mutex.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_mutex.h
--- ./lthread_mutex.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_mutex.h	2016-11-30 18:28:48.000000000 +0900
@@ -35,10 +35,6 @@
 #ifndef LTHREAD_MUTEX_H_
 #define LTHREAD_MUTEX_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include "lthread_queue.h"


@@ -53,8 +49,4 @@ struct lthread_mutex {
 	uint64_t		diag_ref; /* optional ref to user diag data */
 } __rte_cache_aligned;

-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif /* LTHREAD_MUTEX_H_ */
Binary files ./lthread_mutex.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_mutex.o differ
Binary files ./lthread.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread.o differ
diff -uprN ./lthread_objcache.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_objcache.h
--- ./lthread_objcache.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_objcache.h	2016-11-30 18:28:48.000000000 +0900
@@ -33,11 +33,6 @@
 #ifndef LTHREAD_OBJCACHE_H_
 #define LTHREAD_OBJCACHE_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
-
 #include <string.h>

 #include <rte_per_lcore.h>
@@ -159,8 +154,5 @@ _lthread_objcache_free(struct lthread_ob
 }


-#ifdef __cplusplus
-} /* extern C */
-#endif

 #endif				/* LTHREAD_OBJCACHE_H_ */
diff -uprN ./lthread_pool.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_pool.h
--- ./lthread_pool.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_pool.h	2016-11-30 18:28:48.000000000 +0900
@@ -69,10 +69,6 @@
 #ifndef LTHREAD_POOL_H_
 #define LTHREAD_POOL_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <rte_malloc.h>
 #include <rte_per_lcore.h>
 #include <rte_log.h>
@@ -332,8 +328,5 @@ _qnode_pool_destroy(struct qnode_pool *p
 	return 0;
 }

-#ifdef __cplusplus
-} /* extern C */
-#endif

 #endif				/* LTHREAD_POOL_H_ */
diff -uprN ./lthread_queue.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_queue.h
--- ./lthread_queue.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_queue.h	2016-11-30 18:28:48.000000000 +0900
@@ -69,10 +69,6 @@
 #ifndef LTHREAD_QUEUE_H_
 #define LTHREAD_QUEUE_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include <string.h>

 #include <rte_prefetch.h>
@@ -303,8 +299,4 @@ _lthread_queue_remove(struct lthread_que
 }


-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_QUEUE_H_ */
diff -uprN ./lthread_sched.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_sched.h
--- ./lthread_sched.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_sched.h	2016-11-30 18:28:48.000000000 +0900
@@ -62,10 +62,6 @@
 #ifndef LTHREAD_SCHED_H_
 #define LTHREAD_SCHED_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include "lthread_int.h"
 #include "lthread_queue.h"
 #include "lthread_objcache.h"
@@ -153,8 +149,4 @@ void _sched_timer_cb(struct rte_timer *t
 void _sched_shutdown(__rte_unused void *arg);


-#ifdef __cplusplus
-} /* extern C */
-#endif
-
 #endif				/* LTHREAD_SCHED_H_ */
Binary files ./lthread_sched.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_sched.o differ
diff -uprN ./lthread_timer.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_timer.h
--- ./lthread_timer.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_timer.h	2016-11-30 18:28:48.000000000 +0900
@@ -35,10 +35,6 @@
 #ifndef LTHREAD_TIMER_H_
 #define LTHREAD_TIMER_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include "lthread_int.h"
 #include "lthread_sched.h"

@@ -79,8 +75,5 @@ _timer_stop(struct lthread *lt)
 	}
 }

-#ifdef __cplusplus
-} /* extern C */
-#endif

 #endif /* LTHREAD_TIMER_H_ */
diff -uprN ./lthread_tls.h /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_tls.h
--- ./lthread_tls.h	2017-03-14 01:43:01.073260218 +0900
+++ /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_tls.h	2016-11-30 18:28:48.000000000 +0900
@@ -34,11 +34,6 @@
 #ifndef LTHREAD_TLS_H_
 #define LTHREAD_TLS_H_

-#ifdef __cplusplus
-extern "C" {
-#endif
-
-
 #include "lthread_api.h"

 #define RTE_PER_LTHREAD_SECTION_SIZE \
@@ -58,8 +53,5 @@ void _lthread_tls_destroy(struct lthread
 void _lthread_key_pool_init(void);
 void _lthread_tls_alloc(struct lthread *lt);

-#ifdef __cplusplus
-} /* extern C */
-#endif

 #endif				/* LTHREAD_TLS_H_ */
Binary files ./lthread_tls.o and /home/slank/dpdk-16.07/examples/performance-thread/common/lthread_tls.o differ
